### PR TITLE
Added implementation of MultiWalletMultiTable postgres plugin strategy

### PR DESCRIPTION
**Pull request by JoHenrich**
We saw that the strategy MultiWalletMultiTable was prepared in the code but not yet implemented.

So decided to do that because we experienced some major peformance issues with the implemented strategies so far.

In addition we added another Strategy: MultiWalletSplitDatabaseMultiTable which cluster the tables in different databases depending on the id of the wallet. That lead to a performance boost.

All unit test ran successfull and we tested the strategy in production aswell but we still consider it as experimentall because further tests are required to ensure that there are no undetected misconceptions.

Signed-off-by: j.henrich@esatus.com

### CODE REVIEW

**Comment by ianco** [comment]
Hi @JoHenrich  thanks for the PR!  Note that you still have one commit that is missing the DCO signoff

File: libindy/src/api/anoncreds.rs [hunk 1]
```diff
@@ -935,6 +935,7 @@ pub extern fn indy_prover_create_credential_req(command_handle: CommandHandle,
                                                 cred_offer_json: *const c_char,
                                                 cred_def_json: *const c_char,
                                                 master_secret_id: *const c_char,
+                                                device_key: *const c_char,
```
Comment (ianco) [inline libindy/src/api/anoncreds.rs:938]: Is this change related to the new postgres wallet storage scheme?

If not it should be moved to a separate PR

**Review by ianco — CHANGES_REQUESTED** [review]
Overall looks good but I think the anoncreds changes should be in a separate PR.

**Comment by JoHenrich** [comment]
> Hi @JoHenrich thanks for the PR! Note that you still have one commit that is missing the DCO signoff

Hey, thanks for the fast reply. Yes i saw it too, i don`t manage to sign it off. I have another branch where everything is verified. Should i open a new PR and close this one?

File: libindy/src/api/anoncreds.rs (hunk 1 above)
Comment (JoHenrich) [reply libindy/src/api/anoncreds.rs:938]: We will remove the anoncreds changes. They are not related to that PR. Thanks for the hint

**Comment by JoHenrich** [comment]
Closing this PR because there is another one pending with all commits signed off
